server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -305,7 +305,22 @@ ClusterState addIndexTemplateV2(final ClusterState currentState, final boolean c
             throw new IllegalArgumentException("index template [" + name + "] already exists");
         }
 
-        Map<String, List<String>> overlaps = findConflictingV1Templates(currentState, name, template.indexPatterns());
+        Map<String, List<String>> overlaps = findConflictingV2Templates(currentState, name, template.indexPatterns(), true,
+            template.priority());
+        if (overlaps.size() > 0) {
+            String error = String.format(Locale.ROOT, "index template [%s] has index patterns %s matching patterns from " +
+                    "existing templates [%s] with patterns (%s) that have the same priority [%d]",
```

Comment:
I think we can be a little more explicit with what the user needs to do, maybe something like:

```
index template [foo] has index patterns [bar*] matching patterns from existing templates [baz] with patterns (bar*) that have the same priority [5], multiple index templates may not match during index creation, please use a different priority
```

What do you think?

Reply:
Good idea, I'll use that